### ISSUE TITLE
Pushing gets stuck in an infinite loop

### ISSUE THREAD

**Issue by hpsu** [trakt/Plex-Trakt-Scrobbler#253]
One of my libraries fails to push with an infinite loop. Log file continously spews lines like the snippet below:
```
... snip ...
2016-01-05 22:44:36,081 (7fe5ca7fc700) :  INFO (logging_handler:46) - [plugin.pts.action_manager] Ignoring duplicate "add" action (history: {'performed': ['add'], 'sent': ['add']})
2016-01-05 22:44:36,081 (7fe5ca7fc700) :  DEBUG (logging_handler:46) - [plugin.sync.push] watch() - Invalid action for "Duetterna " [195123] (already scrobbled or duplicate action)
2016-01-05 22:44:36,082 (7fe5ca7fc700) :  INFO (logging_handler:46) - [plugin.pts.action_manager] Ignoring duplicate "add" action (history: {'performed': ['add'], 'sent': ['add']})
2016-01-05 22:44:36,082 (7fe5ca7fc700) :  DEBUG (logging_handler:46) - [plugin.sync.push] watch() - Invalid action for "Duetterna " [195124] (already scrobbled or duplicate action)
... snip ...
```

The episode appears to be from a swedish series "Så mycket bättre". I have tried killing the process and restarting the push multiple times, but this always happens when it comes to this series.

I am using version from git cd88f4f6145b9358b53d61a52783b70002a5367a

**Comment by fuzeman** [comment]
I would try update to the beta branch first as it has a completely rebuilt syncing system. It is very close to release, with just some work on the channel menu left (progress/eta calculations) to complete.

**Comment by hpsu** [comment]
Thanks! The situation on the beta branch is actually a bit worse off for me. 

The log shows a ton of lines like below and then hangs until I cancel with 100% CPU usage on one core and no output to log.

I call it worse since it now hangs on all my libraries. I tried fooling around with settings locally and on http://trakt-for-plex.github.io/ to no avail. Any ideas?

```
2016-01-12 17:29:34,426 - plugin.sync.handlers.watched.push (7f70ce7fc700) :  DEBUG (plugin.sync.handlers.watched.push:58) - Episodes.on_added(3, ...) - p_view_count: 1, p_value: datetime.datetime(2014, 10, 7, 19, 57, 4, tzinfo=<UTC>), t_value: None
2016-01-12 17:29:34,666 - plugin.sync.handlers.watched.push (7f70ce7fc700) :  DEBUG (plugin.sync.handlers.watched.push:58) - Episodes.on_added(28, ...) - p_view_count: 1, p_value: datetime.datetime(2014, 10, 7, 19, 57, 7, tzinfo=<UTC>), t_value: None
```

**Comment by fuzeman** [comment]
Might be related to #247, haven't been able to work out what causes this issue though..

------------------

Could you try install this branch: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/247.zip

The ```issue/247``` branch has some extra debug messages that could help debug this issue, so send me the log file *(email to me@dgardiner.net, or post a link here)* after trying another sync (where it hangs, and causes 100% cpu usage).

You can also try switching the "Matcher" option from "Plex Extended" to "Plex" in the Plex/Web channel settings, this will give us an indication of where the sync is getting stuck.

**Comment by hpsu** [comment]
You seem to be right about the matcher. Non-extended seems to work fine. I have mailed the log to you.

I added some own debug info and found the following:
in Libraries/Shared/plugin/sync/modes/push/shows.py:81-ish 

```
                if not p_guid or p_guid.agent not in GUID_AGENTS:
                    mark_unsupported(unsupported_shows, ids['show'], p_guid, p_show)
                    continue

                key = (p_guid.agent, p_guid.sid)
                identifier = (season_num, episode_num)

                # Try retrieve `pk` for `key`
                pk = self.trakt.table.get(key)
                log.debug('#### DEBUG %r %r ####', key, identifier)
```
which yields debug info like:
```
2016-01-13 13:25:54,719 - plugin.sync.modes.push.shows     (7f0d353dd700) :  DEBUG (plugin.sync.modes.push.shows:96) - #### DEBUG ('tvdb', u'280591') (2, 644) ####
2016-01-13 13:25:54,719 - plugin.sync.modes.push.shows     (7f0d353dd700) :  DEBUG (plugin.sync.modes.push.shows:96) - #### DEBUG ('tvdb', u'280591') (2, 645) ####
```
(It continues indefinitely and episode_num increments every time).

The series ID is a swedish show that seems to have blank information in english.
http://thetvdb.com/?tab=series&id=280591&lid=7

**Comment by fuzeman** [comment]
Thanks, looks like the extended matcher must be doing something weird with unicode characters :confused: 

Will try get a fix released today.

**Comment by fuzeman** [comment]
Actually, this might not be an issue with unicode characters..

Could you send me the episode file names? I can then just run them through the matcher and see what's happening.

**Comment by hpsu** [comment]
Thanks for looking into this! I have mailed you the filenames. 

I thought it had to do with thetvdb:s API returning empty data for english (plex has matched the series as english), but looking at
http://thetvdb.com/api/<apikey>/series/280591/all/en.xml,
although many fields are empty, the episode structure is still intact

Tell me if you need me to test something.

**Comment by fuzeman** [comment]
The changes in the [issue/253](https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/253.zip) branch might fix this, haven't had time to actually test this though..

**Comment by hpsu** [comment]
That seems to have fixed the problem for me, thanks! 

Most likely fixes #247 as well.

**Comment by fuzeman** [comment]
Great :smile:, will include this fix in a new beta release later this week.

> Most likely fixes #247 as well.

Yeah, pretty sure this was a duplicate of #247
